### PR TITLE
Add pod pause/unpause

### PR DESCRIPTION
**Pull request by haircommander**
Added Pause() and Unpause() to libpod/pod.go

Added man pages, tests and completions

Signed-off-by: haircommander <pehunt@redhat.com>

### CODE REVIEW

**Comment by baude** [comment]
@rh-atomic-bot r+

**Comment by baude** [comment]
bot, retest this please

File: cmd/podman/pod_unpause.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+			}
+			lastError = errors.Wrapf(err, "unable to find pod %s", i)
+			continue
+		}
+		pods = append(pods, pod)
+	}
+
+	for _, pod := range pods {
+		ctr_errs, err := pod.Unpause()
+		if err != nil {
```
Comment (mheon) [inline cmd/podman/pod_unpause.go]: The logic here is wrong - `err` will always be set if there's an error, even if `ctr_errs` is not nil

File: cmd/podman/pod.go [hunk 2]
```diff
@@ -22,6 +22,8 @@ var (
 			podRmCommand,
 			podStartCommand,
 			podStopCommand,
+			podPauseCommand,
```
Comment (mheon) [inline cmd/podman/pod.go]: Alphabetize

Comment (haircommander) [reply cmd/podman/pod.go]: what should I alphabetize here?

**Comment by mheon** [comment]
LGTM
@rhatdan @TomSweeneyRedHat @umohnani8 PTAL

**Comment by TomSweeneyRedHat** [comment]
All kinds of test unhappiness, including travis @haircommander

File: cmd/podman/pod_unpause.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	"github.com/projectatomic/libpod/libpod"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+)
+
+var (
+	podUnpauseFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "all, a",
+			Usage: "unpause all running pods",
```
Comment (TomSweeneyRedHat) [inline cmd/podman/pod_unpause.go]: Can you pause a running pod?  Should this be "unpause all paused pods"?

File: cmd/podman/pod_unpause.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+   Unpauses one or more pods.  The pod name or ID can be used.
+`
+
+	podUnpauseCommand = cli.Command{
+		Name:                   "unpause",
+		Usage:                  "Unpause one or more pods",
+		Description:            podUnpauseDescription,
+		Flags:                  podUnpauseFlags,
+		Action:                 podUnpauseCmd,
+		ArgsUsage:              "POD-NAME [POD-NAME ...]",
```
Comment (TomSweeneyRedHat) [inline cmd/podman/pod_unpause.go]: Could be Name or ID no?  I think we've other usage's that show you could do one or the other, emulate those.

File: cmd/podman/pod_pause.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+   Pauses one or more pods.  The pod name or ID can be used.
+`
+
+	podPauseCommand = cli.Command{
+		Name:                   "pause",
+		Usage:                  "Pause one or more pods",
+		Description:            podPauseDescription,
+		Flags:                  podPauseFlags,
+		Action:                 podPauseCmd,
+		ArgsUsage:              "POD-NAME [POD-NAME ...]",
```
Comment (TomSweeneyRedHat) [inline cmd/podman/pod_pause.go]: Name or ID no?

**Comment by TomSweeneyRedHat** [comment]
Couple nit things.  
Education question for me.  is there a Pod state?  If so, if we have containers in the pod that are paused and others that are started in the pod, what is the Pod state?

**Comment by mheon** [comment]
@TomSweeneyRedHat We don't really have a solid pod state - we display one in `ps` but generally speaking it's just based on the states its containers are in. I think we would consider a pod Running in your given scenario, given that not all containers are paused?

**Comment by haircommander** [comment]
@TomSweeneyRedHat Yeah, there's documentation on what container state constitutes what pod state in the pod ps man page. @mheon is correct in your scenario, if there are any running containers the pod is considered running.

**Comment by mheon** [comment]
Should be good to merge once travis goes green

**Comment by TomSweeneyRedHat** [comment]
Travis isn't hip @haircommander

**Comment by mheon** [comment]
@haircommander Can you squash the commits down?

**Comment by haircommander** [comment]
yeah I'm trying to figure out how without having to redo the merge again.

**Comment by haircommander** [comment]
There's some issue with the ginkgo suite and this code. It just stalls after finishing. I reproduce locally and am trying to figure out what is making it happen.

**Comment by mheon** [comment]
@haircommander Can you rebase against master to fix the tests?

**Comment by mheon** [comment]
I think this is the last thing I'd like in for today's release.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**